### PR TITLE
Support for logging GraphQL trip requests

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [ ] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Supersedes and improves upon #219. Allows trip requests to be saved with GraphQL parameters.
This PR does break the itinerary checking feature, which is ported to GraphQL in #243.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java [hunk 1]
```diff
@@ -189,15 +190,12 @@ private String proxyPost(Request request, Response response) {
...
-
-                String fromPlace = (String) graphQlVariables.get("fromPlace");
-                String toPlace = (String) graphQlVariables.get("toPlace");
+                OtpGraphQLVariables graphQlVariables = getPOJOFromJSON(requestBody, OtpGraphQLQuery.class).variables;
 
                 // Follows the method used in otp-ui core-utils storage.js
                 String randomBatchId = Integer.toString((int) (Math.random() * 1_000_000_000), 36);
 
-                if(!handlePlanTripResponse(randomBatchId, fromPlace, toPlace, otpDispatcherResponse, otpUser)) {
+                if(!handlePlanTripResponse(randomBatchId, graphQlVariables, otpDispatcherResponse, otpUser)) {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java:198]: Nit. Space after "if".

File: src/main/java/org/opentripplanner/middleware/models/TripRequest.java [hunk 2]
```diff
@@ -61,6 +68,20 @@ public TripRequest(
         this.requestParameters = requestParameters;
     }
 
+    public TripRequest(
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/TripRequest.java:71]: The other constructor is only used in a unit test and this is only used in code. Consider removing the other constructor and updating the unit test to use this constructor.

File: src/main/java/org/opentripplanner/middleware/otp/OtpGraphQLQuery.java
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/otp/OtpGraphQLQuery.java:1]: Consider putting these OtpGraphQL... classes into their own "graphql" package and then perhaps drop "OtpGraphQL" from the class name?

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package org.opentripplanner.middleware.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:6]: Replace with explicit classes.

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * Location of the GraphQL plan query template file, as Java resource.
+     */
+    public static final String PLAN_QUERY_RESOURCE = "queries/planQuery.graphql";
+
+    /**
+     * Return the full GraphQL plan file planQueryTemplate in Java string format, with {@code "} as {@code \"}
+     */
+    public static String getPlanQueryTemplate() {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:22]: This method does not appear to be called which suggests that the plan query is not loaded. This class also isn't referenced anywhere that I can see, is it needed?

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
+package org.opentripplanner.middleware.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+
+public class GraphQLUtils {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:8]: SonarLint: Add a private constructor to hide the implicit public one.

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            GraphQLUtils.planQueryTemplate = planQueryTemplateAsString(PLAN_QUERY_RESOURCE);
+	}
+        return GraphQLUtils.planQueryTemplate;
+    }
+
+    /**
+     * Return a GraphQL planQueryTemplate in Java string format, with {@code "} as {@code \"}
+     * @param resource the plan file or any GraphQL file
+     */
+    static String planQueryTemplateAsString(String resource) {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:33]: SonarLint: Remove the resource method parameter and refer to `GraphQLUtils.PLAN_QUERY_RESOURCE` directly.

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        StringBuilder builder = new StringBuilder();
+        try (var reader = new BufferedReader(new InputStreamReader(
+            GraphQLUtils.class.getClassLoader().getResourceAsStream(resource)
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:36]: SonarLint: Suggestion `Objects.requireNonNull(GraphQLUtils.class.getClassLoader().getResourceAsStream(GraphQLUtils.PLAN_QUERY_RESOURCE))`

File: src/main/resources/queries/planQuery.graphql
Comment (br648) [inline src/main/resources/queries/planQuery.graphql:1]: I'm concerned about the size of this query and maintaining it. I'm not sure it is even used? Perhaps in `GraphQLUtils` explain how this was produced and how\where it is used?

**Review by br648 — CHANGES_REQUESTED** [review]
A few things to address. Also there are no unit tests covering graphQL, are these to following in another PR?

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java (hunk 4 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:22]: You may be right, the bulk of this PR deals with the result of the call to the GraphQL endpoint, not the call itself. Totally missed that. In that case, should I remove the code that breaks the itinerary checks?

File: src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java:198]: Fixed in https://github.com/ibi-group/otp-middleware/pull/245/commits/473d3fac1725a39dd7c74751c405fd90054fc086.

File: src/main/java/org/opentripplanner/middleware/models/TripRequest.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/models/TripRequest.java:71]: Fixed in https://github.com/ibi-group/otp-middleware/pull/245/commits/074bf552d196c488f114ae7a343f1742fdda3161 in #245

File: src/main/java/org/opentripplanner/middleware/otp/OtpGraphQLQuery.java
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/otp/OtpGraphQLQuery.java:1]: Updated in https://github.com/ibi-group/otp-middleware/pull/245/commits/31a141a37d5d8cc1594aaaa7ad3c025d4a2f675f

**Comment by binh-dam-ibigroup** [comment]
Closing in favor of #245.

**Pull request closed without merging**